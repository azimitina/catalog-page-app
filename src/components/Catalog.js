import React from "react";
import Cards from "./Cards";
import get from "lodash/get";
import { Grid } from "semantic-ui-react";
import "./Catalog.scss";

function Catalog({ data }) {
  return (
    <Grid stackable columns={4}>
      {data.map((item) => {
        return (
          <Grid.Column key={item.sku}>
            <Cards
              name={item._embedded.brand.name}
              meta={item.name}
              price={item.price}
              final_price={item.final_price}
              marketingMessaging={get(
                item,
                "messaging.marketing[0].short",
                null
              )}
              image={get(item, "_embedded.images[0].thumbnail", null)}
            />
          </Grid.Column>
        );
      })}
    </Grid>
  );
}

export default Catalog;
