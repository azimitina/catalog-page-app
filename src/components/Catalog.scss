.ui.grid {
  margin: 30px 0 !important;
}

.ui.stackable.four.column.grid {
  border: 1px solid #f0f0f0;
  margin: 10px 0;
  padding: 0;
  min-height: 500px;
}
