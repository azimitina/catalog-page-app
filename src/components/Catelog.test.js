import { shallow } from "enzyme";
import Catalog from "./Catalog";

describe("Catalog", () => {
  it("should render Catalog", () => {
    const props = {
      data: [
        {
          sku: "123456",
          _embedded: { brand: { name: "Nike" } },
          name: "Indy Legging",
          price: 300,
          final_price: 300,
        },
      ],
    };
    const wrapper = shallow(<Catalog {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should render Catalog with multiple cards", () => {
    const props = {
      data: [
        {
          sku: "123456",
          _embedded: { brand: { name: "Nike" } },
          name: "Indy Legging",
          price: 300,
          final_price: 300,
        },
        {
          sku: "34567",
          _embedded: {
            brand: { name: "Nike" },
            images: [
              {
                thumbnail: "nike-small.jpg",
              },
            ],
          },
          name: "Indy Legging",
          price: 300,
          final_price: 300,
          messaging: { marketing: [{ short: "Just do it" }] },
        },
      ],
    };
    const wrapper = shallow(<Catalog {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
