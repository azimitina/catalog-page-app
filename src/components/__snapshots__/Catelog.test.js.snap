// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Catalog should render Catalog 1`] = `
<Grid
  columns={4}
  stackable={true}
>
  <GridColumn
    key="123456"
  >
    <Cards
      final_price={300}
      image={null}
      marketingMessaging={null}
      meta="Indy Legging"
      name="Nike"
      price={300}
    />
  </GridColumn>
</Grid>
`;

exports[`Catalog should render Catalog with multiple cards 1`] = `
<Grid
  columns={4}
  stackable={true}
>
  <GridColumn
    key="123456"
  >
    <Cards
      final_price={300}
      image={null}
      marketingMessaging={null}
      meta="Indy Legging"
      name="Nike"
      price={300}
    />
  </GridColumn>
  <GridColumn
    key="34567"
  >
    <Cards
      final_price={300}
      image="nike-small.jpg"
      marketingMessaging="Just do it"
      meta="Indy Legging"
      name="Nike"
      price={300}
    />
  </GridColumn>
</Grid>
`;
